src/main/java/jenkins/advancedqueue/jobinclusion/strategy/FolderBasedJobInclusionStrategy.java

```diff
@@ -62,9 +67,32 @@ public ListBoxModel getListFolderItems() {
 
 	};
 
+	static public class JobPattern {
+		private String jobPattern;
+
+		@DataBoundConstructor
+		public JobPattern(String jobPattern) {
+			this.jobPattern = jobPattern;
+		}
+
+	}
+
 	private String folderName;
 
+	private boolean useJobFilter = false;
+
+	private String jobPattern = ".*";
+	private transient Pattern compiledPattern;
+
 	@DataBoundConstructor
+	public FolderBasedJobInclusionStrategy(String folderName, JobPattern jobFilter) {
+		this.folderName = folderName;
+		this.useJobFilter = (jobFilter != null);
+		if (this.useJobFilter) {
+			this.jobPattern = jobFilter.jobPattern;
```

Comment:
Empty pattern from config will override the default value. I would rather make the field nullable, get rid of the default, add `@CheckForNull` annotation to getter methods. Once the value is null, you can just skip regex check to get a better performance